.github/ISSUE_TEMPLATE/breaking_change.md

```diff
@@ -0,0 +1,33 @@
+---
+name: Breaking Change
+about: Track plans for a breaking change
+title: "Removal of ? in v2."
+labels: "breaking change"
+assignees: ""
+---
+
+## Broken API or pattern
+
+<!-- Name the APIs involved specifically, etc. -->
+
+**Preparation References**
+
+<!-- A URL to the deprecation release notes. Alternatively, describe actions required where broken API or pattern is in use. -->
+
+**Packages**
```

Comment:
this seems duplicative of what will be in the change set for the break itself

Reply:
can we reuse that changeset for much of this, it feels quite redundant to repeat the same things in different places and formats

Reply:
The release notes URL is the way to avoid redundancy. There should already be a changeset from deprecation changes that have made it to a release. ...unless you are really pushing the envelope. In the envelope pushing case, it would be okay to edit the issue after release or copy the content from the changeset.

Reply:
+1 on this section in particular - typing out the list of packages feels very tedious, when it's already captured in the changeset.

Reply:
I think I misread the initial comment. I don't know how to get GitHub to reference a single line. I focused on the deprecation notes section, but I suspect this is for packages.
`Packages` is here because we want a search hit and we want easy collection/survey of impacted packages.
Remember this is a service we are providing to customers. (Also, part of a commitment for the legacy break process.)
The impacted packages should normally be one or two. Doesn't seem like a big hit.
Alternatively, if all of the information is readily gatherable, then perhaps write script to generate this.

Reply:
> this is a service we are providing to customers

I see 3 different information sources here:

1. The stream of past release notes covering a variety of "upcoming" breaks (deprecations, etc)
2. These sub-issues
3. The release notes for the 2.x0 release covering all the breaks.

Stepping back to discuss the value prop of (2) as a whole -- During what time period would a consumer review this issue, and learn something new or do something different, compared to existing workflows around (1)?  In other words, I expect that a highly engaged partner is reading all release notes for every minor, and taking action to prepare for the future.  In that case, this issue will be a checksum, ideally a no-op.  Same with (3).

So the value prop seems to be -- we're putting in extra work to distill and refactor (1) as a way to peer into the future and see what (3) _will be_ - while there's still time to do something about it without delaying the bump.

> Doesn't seem like a big hit.

I see that, but I also feel the resistance to it.  When do I ever type a package name?  I don't - Imports happen automatically, the changeset tool gives me a list to choose from, etc.

Hopefully that helps clarify what I'm seeing / bringing to this, for whatever it's worth.

> if all of the information is readily gatherable, then perhaps write script to generate this.

Not sure how this would be accomplished without updatine changeset semantics to include the future version where a break will happen (in the changeset announcing it).  But then what if plans change?  Seems tricky.

Reply:
Probably less useful for the very active consumers, but I am not sure how many of those we have. There are a lot deprecated API uses in our codebase and some consumers. There are a number of things get deprecated with an expected timeframe and we slip far past that. In the 2.20 round there is at least one very old deprecation.
So, these (2) make up the we-are-serious notifications and are probably the most important of the streams. The third stream being the release where the breaks happen is too late for our purposes.
I think you might be underestimating how lazy because we are busy software engineers are.

I feel the resistance. It is hard to measure the needs of customers until we do something more uniform. It is also hard to see how hard it is to handle this request without trying. I feel like per the prior section the release notes / change set should be on-hand and you just need a little copy-paste.

Part of the point of suggesting automation is that if it is hard to figure out, then (assuming the information is useful in the first place) it we should make it easy.

### Is the information useful?
When I see a random API name, I don't know what package it is from often. As a consumer with API name and package name I can make enough links in my brain to have a really good guess if my stuff is impacted. `Packages` are the high-order bit and could make sense to stick in the title, but that is just too much. I debated placing `Packages` before `Preparation References`, but I felt like people who didn't recognize the API could scroll a bit and those who do and will have work to do get to the stuff they want faster (shorten the longer pole).

Reply:
Definitely willing to try it!  As long as we are collecting feedback from those who use this workflow (or that don't!)

Reply:
Yeah - I don't think I can get the feedback until 2.30+ since there is a lot of variety in 2.20 without this simplified template in place.